internal/cli/cli.go

```diff
@@ -71,6 +73,19 @@ func Run(ctx context.Context, io *iostreams.IOStreams, args ...string) int {
 	cmd := root.New()
 	cmd.SetOut(io.Out)
 	cmd.SetErr(io.ErrOut)
+
+	// Special case for the launch command, support `flyctl launch args -- [subargs]`
+	// Where the arguments after `--` are passed to the scanner/dockerfile generator.
+	// This isn't supported natively by cobra, so we have to manually split the args
+	// See: https://github.com/spf13/cobra/issues/739
+	if len(args) > 0 && args[0] == "launch" {
+		index := slices.Index(args, "--")
+		if index >= 0 {
+			ctx = flag.WithExtraArgs(ctx, args[index+1:])
+			args = args[0:index]
```

Comment:
Minor nit: omitting the `0` is typically more common (e.g. `args[:index]`)

Reply:
Will fix on my next commit (as well as remove the leftover debug statement) after we come to agreement on error handling.

Reply:
Fixed